src/package-url.js

```diff
@@ -57,7 +57,7 @@ class PackageURL {
   }
 
   _handlePyPi() {
-    this.name = this.name.toLowerCase().replace('_', '-');
+    this.name = this.name.toLowerCase().replace(/_/g, '-');
```

Comment:
Looking at this with @declanhunter , we discovered (and I had no clue about this!) that `replace` with a `substring` only replaces the first occurrence. `replaceAll` doesn't appear to be available in `node`, so this `regex` solution for `replace` seemed the most broadly acceptable solution